packages/@aws-cdk/core/lib/arn.ts

```diff
@@ -268,3 +284,45 @@ function parseToken(arnToken: string, sep: string = '/', hasName: boolean = true
     return { partition, service, region, account, resource, resourceName, sep };
   }
 }
+
+
+/**
+ * Validate that a string is either unparseable or looks mostly like an ARN
+ */
+function parseArnShape(arn: string): 'unparseable' | string[] {
+  const components = arn.split(':') as Array<string>;
+  const looksLikeArn = arn.startsWith('arn:') && components.length >= 6 && components.length <= 7;
+  if (Token.isUnresolved(arn) && !looksLikeArn) {
+    return 'unparseable';
+  }
+  // If the ARN merely contains Tokens, but otherwise *looks* mostly like an ARN,
+  // it's a string of the form 'arn:${partition}:service:${region}:${account}:abc/xyz'.
+  // Parse fields out to the best of our ability.
+  // Tokens won't contain ":", so this won't break them.
+
+  if (components.length < 6) {
+    throw new Error('ARNs must have at least 6 components: ' + arn);
+  }
+
+  // If the ARN merely contains Tokens, but otherwise *looks* mostly like an ARN,
+  // it's a string of the form 'arn:${partition}:service:${region}:${account}:abc/xyz'.
+  // Parse fields out to the best of our ability.
+  // Tokens won't contain ":", so this won't break them.
+  const [arnPrefix, , service, , , sixth] = components;
+
+  if (arnPrefix !== 'arn') {
+    throw new Error('ARNs must start with "arn:": ' + arn);
+  }
+
+  // partition is not required, 'cognito-sync' doesn't use it.
```

Comment:
?

Reply:
Going off of this test, assuming it's accurate: https://github.com/aws/aws-cdk/blob/huijbers/fix-resource-path2/packages/@aws-cdk/core/test/arn.test.ts#L160

Reply:
cognito sync ARN should include the partition? https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazoncognitosync.html#amazoncognitosync-resources-for-iam-policies